test/e2e/framework/framework.go

```diff
@@ -284,8 +286,25 @@ func (f *Framework) AfterEach() {
 		if !f.SkipNamespaceCreation {
 			DumpAllNamespaceInfo(f.ClientSet, f.Namespace.Name)
 		}
-		By(fmt.Sprintf("Dumping a list of prepulled images on each node"))
-		LogContainersInPodsWithLabels(f.ClientSet, metav1.NamespaceSystem, ImagePullerLabels, "image-puller", Logf)
+
+		logFunc := Logf
+		if TestContext.ReportDir != "" {
+			filePath := path.Join(TestContext.ReportDir, "image-puller.txt")
+			By(fmt.Sprintf("Dumping a list of prepulled images on each node to file %v", filePath))
+			file, err := os.Create(filePath)
+			if err != nil {
+				Logf("Failed to create a file with image-puller data %v: %v", filePath, err)
+				return
```

Comment:
Change to else - otherwise you will loose everything that is running later.
[The same you did above]